diff --git a/ouster_client/include/ouster/impl/open_source_impl.h b/ouster_client/include/ouster/impl/open_source_impl.h
index aff9a6e..fb43b50 100644
--- a/ouster_client/include/ouster/impl/open_source_impl.h
+++ b/ouster_client/include/ouster/impl/open_source_impl.h
@@ -10,6 +10,7 @@
 #include <stdexcept>
 #include <string>
 #include <vector>
+#include <stdexcept>
 
 #include "ouster/io_type.h"
 #include "ouster/scan_source_utils.h"
diff --git a/ouster_client/src/io_type.cpp b/ouster_client/src/io_type.cpp
index f48f019..849c552 100644
--- a/ouster_client/src/io_type.cpp
+++ b/ouster_client/src/io_type.cpp
@@ -10,6 +10,7 @@
 #include <nonstd/optional.hpp>
 #include <stdexcept>
 #include <string>
+#include <stdexcept>
 
 namespace ouster {
 namespace sdk {
