diff --git a/ouster_library/CMakeLists.txt b/ouster_library/CMakeLists.txt
index 3309f61..56b8ae2 100644
--- a/ouster_library/CMakeLists.txt
+++ b/ouster_library/CMakeLists.txt
@@ -59,6 +59,14 @@ function(ouster_library_common _TARGET)
     message("Building Library: ${_TARGET}: Adding ouster_viz")
     message("Building Library: ${_TARGET}: VIZ_INCLUDE_DIRS: ${_INCLUDE_DIRS}")
   endif()
+  if(BUILD_MAPPING)
+    target_link_libraries(${_TARGET} PRIVATE ouster_mapping)
+    get_target_property(_INCLUDE_DIRS ouster_mapping INTERFACE_INCLUDE_DIRECTORIES)
+    target_include_directories(${_TARGET}
+      PUBLIC
+      $<BUILD_INTERFACE:${_INCLUDE_DIRS}>)
+    message("Building Library: ${_TARGET}: Adding ouster_mapping")
+  endif()
 
   add_library(OusterSDK::${_TARGET} ALIAS ${_TARGET})
 endfunction()
@@ -83,6 +91,9 @@ if(BUILD_SHARED_LIBRARY)
   if(TARGET ouster_sensor)
   target_sources(shared_library PRIVATE $<TARGET_OBJECTS:ouster_sensor>)
   endif()
+  if(TARGET ouster_mapping)
+      target_sources(shared_library PRIVATE $<TARGET_OBJECTS:ouster_mapping>)
+  endif()
   ouster_library_common(shared_library)
   target_compile_definitions(shared_library PRIVATE BUILD_SHARED_LIBS_EXPORT)
 endif()
