diff --git a/CMakeLists.txt b/CMakeLists.txt
index b827394..f36f82d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,16 +25,6 @@ endif()
 option(BASALT_BUILD_SHARED_LIB "Build basalt core library as shared" ON)
 option(BASALT_FULLY_STATIC_BINARIES "Link executables fully static" ON)
 
-if(NOT CXX_MARCH)
-  set(CXX_MARCH native)
-endif()
-
-if(NOT APPLE OR NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
-  set(BASALT_MARCH_FLAGS "-march=${CXX_MARCH}")
-else()
-  message(STATUS "Running on Apple ${CMAKE_SYSTEM_PROCESSOR}. Disabled -march flag.")
-endif()
-
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
@@ -119,7 +109,7 @@ else()
   endif()
 endif()
 
-if(APPLE)
+if(0)
   set(CMAKE_FIND_FRAMEWORK LAST)
 
   if(CMAKE_SYSTEM_VERSION VERSION_LESS 19.0.0)
@@ -153,8 +143,6 @@ else()
   message(WARNING "Only Linux and macOS are currently supported")
 endif()
 
-set(BASALT_CXX_FLAGS "${BASALT_CXX_FLAGS} -DEIGEN_DONT_PARALLELIZE")
-
 option(BASALT_INSTANTIATIONS_DOUBLE "Instantiate templates for Scalar=double." ON)
 option(BASALT_INSTANTIATIONS_FLOAT "Instantiate templates for Scalar=float." ON)
 
@@ -179,14 +167,12 @@ endif()
 find_package(Boost CONFIG REQUIRED COMPONENTS date_time filesystem format program_options regex thread)
 find_package(lz4 CONFIG REQUIRED)
 find_package(fmt CONFIG REQUIRED)
-find_package(CLI11 CONFIG REQUIRED)
 find_package(magic_enum CONFIG REQUIRED)
 find_package(nlohmann_json CONFIG REQUIRED)
 find_package(opengv CONFIG REQUIRED)
 find_package(rosbag CONFIG REQUIRED)
 find_package(Pangolin CONFIG REQUIRED)
 find_package(basalt-headers CONFIG REQUIRED)
-find_package(GTest CONFIG REQUIRED)
 
 get_target_property(BASALT_EIGEN_INCLUDE_DIRS Eigen3::Eigen INTERFACE_INCLUDE_DIRECTORIES)
 if(NOT BASALT_EIGEN_INCLUDE_DIRS)
@@ -230,13 +216,7 @@ set_property(TARGET basalt::opencv PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${Open
 set_property(TARGET basalt::opencv PROPERTY INTERFACE_LINK_LIBRARIES ${OpenCV_LIBS})
 
 # Build bundled apriltag library (needs basalt::opencv above)
-add_subdirectory(thirdparty/apriltag EXCLUDE_FROM_ALL)
-
-add_library(basalt::cli11 INTERFACE IMPORTED)
-set_property(TARGET basalt::cli11 PROPERTY INTERFACE_LINK_LIBRARIES CLI11::CLI11)
-
-add_library(basalt::magic_enum INTERFACE IMPORTED)
-set_property(TARGET basalt::magic_enum PROPERTY INTERFACE_LINK_LIBRARIES magic_enum::magic_enum)
+add_subdirectory(thirdparty/apriltag)
 
 set(BASALT_PANGO_TARGETS ${Pangolin_LIBRARIES})
 
@@ -344,17 +324,22 @@ target_sources(basalt
     ${CMAKE_CURRENT_SOURCE_DIR}/src/vi_estimator/sqrt_keypoint_vio.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/src/vi_estimator/sqrt_keypoint_vo.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/src/vi_estimator/vio_estimator.cpp
+    $<TARGET_OBJECTS:apriltag>
 )
 
 target_link_libraries(basalt
   PUBLIC ${STD_CXX_FS} Eigen3::Eigen basalt::opencv basalt::basalt-headers TBB::tbb
-  PRIVATE basalt::magic_enum rosbag apriltag opengv nlohmann_json::nlohmann_json fmt::fmt)
+  PRIVATE basalt::magic_enum rosbag opengv nlohmann_json::nlohmann_json fmt::fmt)
 target_include_directories(basalt PUBLIC
   ${CMAKE_CURRENT_SOURCE_DIR}/include
+  ${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/apriltag/include
   ${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/ros/include)
 target_compile_definitions(basalt PUBLIC ${BASALT_COMPILE_DEFINITIONS})
 #target_compile_definitions(basalt PUBLIC BASALT_DISABLE_ASSERTS)
 
+if(BASALT_BUILD_TOOLS)
+find_package(CLI11 CONFIG REQUIRED)
+
 add_executable(basalt_calibrate src/calibrate.cpp src/calibration/cam_calib.cpp)
 target_link_libraries(basalt_calibrate basalt ${BASALT_PANGO_TARGETS} basalt::cli11)
 basalt_add_gui_link_options(basalt_calibrate)
@@ -465,6 +450,22 @@ if(realsense2_FOUND)
       ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
 endif()
 
+if(BASALT_USE_CHOLMOD)
+  find_package(CHOLMOD REQUIRED)
+  target_link_libraries(basalt PUBLIC SuiteSparse::CHOLMOD)
+  target_compile_definitions(basalt PUBLIC BASALT_USE_CHOLMOD)
+endif()
+
+endif(BASALT_BUILD_TOOLS)
+
+install(TARGETS basalt
+        EXPORT BasaltTargets
+        RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
+        LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
+
+install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+
 file(GLOB SCRIPTS_TO_INSTALL "${CMAKE_CURRENT_SOURCE_DIR}/scripts/basalt_*.py")
 install(PROGRAMS ${SCRIPTS_TO_INSTALL} DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 
@@ -474,5 +475,7 @@ install(FILES ${CONFIG_FILES}
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/data/udev"
   DESTINATION ${CMAKE_INSTALL_PREFIX}/etc/basalt)
 
-
-add_subdirectory(test)
+if(BUILD_TESTING)
+  find_package(GTest CONFIG REQUIRED)
+  add_subdirectory(test)
+endif()
diff --git a/thirdparty/apriltag/CMakeLists.txt b/thirdparty/apriltag/CMakeLists.txt
index 9c36fa8..d1b659a 100644
--- a/thirdparty/apriltag/CMakeLists.txt
+++ b/thirdparty/apriltag/CMakeLists.txt
@@ -1,25 +1,8 @@
-cmake_minimum_required(VERSION 3.2...3.18)
-
-
-SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
-SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
-
-#file(GLOB APRILTAG_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/apriltag-2016-12-01/*.c" "${CMAKE_CURRENT_SOURCE_DIR}/apriltag-2016-12-01/common/*.c")
-#include_directories(apriltag-2016-12-01)
-
-include_directories(${OpenCV_INCLUDE_DIR})
-
 file(GLOB APRILTAG_SRCS "ethz_apriltag2/src/*.cc")
-include_directories(ethz_apriltag2/include)
-
-
-include_directories(../../include)
-include_directories(../basalt-headers/include)
-include_directories(../basalt-headers/thirdparty/Sophus)
-
-add_library(apriltag STATIC ${APRILTAG_SRCS} include/basalt/utils/apriltag.h src/apriltag.cpp)
 
-target_include_directories(apriltag PUBLIC include)
-target_link_libraries(apriltag PUBLIC basalt::opencv Sophus::Sophus)
+add_library(apriltag OBJECT ${APRILTAG_SRCS} include/basalt/utils/apriltag.h src/apriltag.cpp)
 
+target_include_directories(apriltag PRIVATE ${PROJECT_SOURCE_DIR}/include ${CMAKE_CURRENT_SOURCE_DIR}/ethz_apriltag2/include)
+target_include_directories(apriltag PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
+target_link_libraries(apriltag PUBLIC basalt::basalt-headers basalt::opencv Sophus::Sophus)
 
