--- randutils.c
+++ randutils.c
@@ -6,14 +6,24 @@
  * This file may be redistributed under the terms of the
  * GNU Lesser General Public License.
  */
+
+#if defined(HAVE_CONFIG_H)
+#include "config.h"
+#endif
+
 #include <stdio.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <stdlib.h>
 #include <string.h>
+
+#if defined(HAVE_SYS_TIME_H)
 #include <sys/time.h>
+#endif

+#if defined(HAVE_SYS_SYSCALL_H)
 #include <sys/syscall.h>
+#endif

 #include "randutils.h"

