--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,9 +70,6 @@
   add_definitions ( -D_CRT_SECURE_NO_WARNINGS )
 endif()
 
-# Need to set PIC to allow creating shared libraries from object file libraries.
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-
 # Build the set of objects that do not need to be compiled with flags to enable
 # particular architecture features.
 add_library( ${ly_lib_name}_common_objects OBJECT ${ly_common_source_files} )
@@ -170,15 +167,7 @@
 endif()
 
 # this creates the static library (.a)
-add_library( ${ly_lib_static} STATIC ${ly_lib_parts})
-
-# this creates the shared library (.so)
-add_library( ${ly_lib_shared} SHARED ${ly_lib_parts})
-set_target_properties( ${ly_lib_shared} PROPERTIES OUTPUT_NAME "${ly_lib_name}" )
-set_target_properties( ${ly_lib_shared} PROPERTIES PREFIX "lib" )
-if(WIN32)
-  set_target_properties( ${ly_lib_shared} PROPERTIES IMPORT_PREFIX "lib" )
-endif()
+add_library( ${ly_lib_static} ${ly_lib_parts})
 
 # this creates the cpuid tool
 add_executable      ( cpuid ${ly_base_dir}/util/cpuid.c )
@@ -192,11 +181,20 @@
 add_executable      ( yuvconstants ${ly_base_dir}/util/yuvconstants.c )
 target_link_libraries  ( yuvconstants ${ly_lib_static} )
 
-find_package ( JPEG )
-if (JPEG_FOUND)
-  include_directories( ${JPEG_INCLUDE_DIR} )
-  target_link_libraries( ${ly_lib_shared} ${JPEG_LIBRARY} )
-  add_definitions( -DHAVE_JPEG )
+include(CheckFunctionExists)
+check_function_exists(round HAVE_MATH_SYSTEM)
+if(NOT HAVE_MATH_SYSTEM)
+  target_link_libraries(yuvconstants m)
+endif()
+
+option(LIBYUV_WITH_JPEG "Build libyuv with jpeg" ON)
+if (LIBYUV_WITH_JPEG)
+  find_package(JPEG REQUIRED)
+  target_link_libraries(${ly_lib_static} JPEG::JPEG )
+  target_link_libraries(${ly_lib_name}_common_objects JPEG::JPEG )
+  target_compile_definitions(${ly_lib_static} PRIVATE HAVE_JPEG)
+  target_compile_definitions(yuvconvert PRIVATE HAVE_JPEG)
+  target_compile_definitions(${ly_lib_name}_common_objects PRIVATE HAVE_JPEG)
 endif()
 
 if(UNIT_TEST)
@@ -242,10 +240,6 @@
 
 
 # install the conversion tool, .so, .a, and all the header files
-install ( TARGETS yuvconvert DESTINATION bin )
-install ( TARGETS ${ly_lib_static}						DESTINATION lib )
-install ( TARGETS ${ly_lib_shared} LIBRARY DESTINATION lib RUNTIME DESTINATION bin ARCHIVE DESTINATION lib )
+install ( TARGETS yuvconvert yuvconstants DESTINATION bin)
+install ( TARGETS ${ly_lib_static} RUNTIME DESTINATION bin ARCHIVE DESTINATION lib LIBRARY DESTINATION lib)
 install ( DIRECTORY ${PROJECT_SOURCE_DIR}/include/		DESTINATION include )
-
-# create the .deb and .rpm packages using cpack
-include ( CM_linux_packages.cmake )
