--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,7 @@
     unset(TREE_SITTER_ABI_VERSION CACHE)
     message(FATAL_ERROR "TREE_SITTER_ABI_VERSION must be an integer")
 endif()
-
+if(0)
 find_program(TREE_SITTER_CLI tree-sitter DOC "Tree-sitter CLI")

 add_custom_command(OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/src/parser.c"
@@ -23,7 +23,7 @@
                             --abi=${TREE_SITTER_ABI_VERSION}
                    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
                    COMMENT "Generating parser.c")
-
+endif()
 add_library(tree-sitter-sql src/parser.c)
-if(EXISTS src/scanner.c)
+if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/src/scanner.c")
   target_sources(tree-sitter-sql PRIVATE src/scanner.c)
@@ -41,7 +41,6 @@
 set_target_properties(tree-sitter-sql
                       PROPERTIES
                       C_STANDARD 11
-                      POSITION_INDEPENDENT_CODE ON
                       SOVERSION "${TREE_SITTER_ABI_VERSION}.${PROJECT_VERSION_MAJOR}"
                       DEFINE_SYMBOL "")

